### ISSUE TITLE
Archiving or downgrading quality after a period of time

### ISSUE THREAD

**Issue by codemann8** [Sonarr/Sonarr#2162]
I can only imagine other peoples' growing libraries, but mine is growing too fast.  I need a way to be able to take older seasons of shows and downgrading to a lower quality over time to save on disk space.  My 6TB RAID5 is fast approaching full.  This would be such a highly desired feature and I think the community would agree.

I'd envision the design like this:
-Each series would have an "Archive quality profile" to determine which quality to switch to.
-Each series would have an integer value to determine the number of days (after air date) in which it should wait until downgrading the quality.
-As a part of the normal scheduled downloading routine, check which episodes are marked for archive status (by date checking) and download new quality if they don't meet the archiving profile quality.

**Comment by markus101** [comment]
You can re-order a profile to rank lower quality releases above higher quality ones to achieve downgrading, which would work well if you're looking to downgrade to `DVD` as they wouldn't be grabbed right away since they don't exist immediately. This wouldn't work with SDTV/WEBDL-480p because they would be released around the same time as the original release.

> As a part of the normal scheduled downloading routine, check which episodes are marked for archive status (by date checking) and download new quality if they don't meet the archiving profile quality.

This would only work if the lower quality releases are re-posted/re-released, since that is how Sonarr finds releases and subsequent upgrades (as explained in the FAQ).

I think a second profile having to be linked to every series would be clunky and confusing for most users, offhand I'm not sure if there is a better way to achieve this, but if it's something we choose to do, I think we'd want something else to achieve it.

Personally I don't see this being useful for most users, since downgrading would be too dependent on releases being republished or manually searching for items later on, but I'll let @Taloth weigh in.

**Comment by codemann8** [comment]
I mean, of course I know there's a manual way to achieve this.  But, the whole point of Sonarr is automation.  I'm just saying there is a fundamental issue with disk space inevitably getting full and I think every user can relate to that.  I have 6TB of space myself, I can't imagine too many people having very much more than that but I'm running out of space quickly, so I would think many other users have to be running into space issues as well.  The scenario you mentioned that I couldn't do is exactly what I'd want.  Initially, I'd want to download the HDTV/1080p versions, but perhaps 90 days later I want to replace it with the SDTV/480p version.  This type of functionality shouldn't depend on when these are released, it would need to be an active search.  An alternative to active searching (to minimize API calls to indexers) could be to cache all the releases for any TV shows that are set up for Archiving, then when the time hits, you reference the cache instead of doing an active search.

**Comment by kyse** [comment]
I'd say the u unmonitor feature is basically archiving as far as sonarr is concerned.  However adding some automated timing mechanism would be an improvement.  However for redownloading the file in a lower resolution, might present some issues.  No longer available at a later time is probably a good possibility  (not enough blocks), wasted bandwidth, etc.  Would be cool to use ffmpeg to switch up to an acceptable archive size though.  That might be out of the scope sonarr wants to handle though.

**Comment by charliwest** [comment]
I tried to do as you suggested and for a series that I already have 720p downloads for I created an archive profile, with dvd and web-420p as the only options. In the series when I look manually it shows no availability 
Release Rejected
Existing file meets cutoff: WEBDL-480p
Quality for existing file on disk is of equal or higher preference: HDTV-720p v1

How can I ge it to download the lower quality as you suggested please?

**Comment by gibxxi** [comment]
What I've done to mitigate this very issue (manually) is to use the new(ish) dbPoweramp Video Converter software to convert all shows I have that are not only complete, but also at the required 'final' quality / resolution level from the myriad of different formats (*.AVI, *.MKV, *.AVI, etc.) and codecs (DivX, Xvid, WMV, H.264 / X264, etc.) to HEVC - X265. Now while it's true that HEVC requires dedicated hardware decoding, it also achieves HALF the filesize of the original, discounting the audio codec used.

I also converted any audio tracks down to HE-AAC stereo. Now in terms of audio, from a personal standpoint, I don't have much intrest in my TV shows being in Dolby Digital or DTS multi-channel format, precisely because *it IS a tv-show*. So I understand that others won't want to do that. But dBpoweramp can preserve the audio from the source file(s).

I managed to half the space usage on one of my NAS devices (which houses my tv-show library) so the gains can be made. The only caveat here is, the process is dog-slow, and resouce heavy. So While i'd like to see something like this available in Sonarr, I don't think my Celeron-powered NAS has the guts to take on the task. It is however an alternative option, and if you don't want to go down the dBpoweramp route, there is other software out there that will accomplish the same end-result. It will just require some time on your part to get to that point.

Gib.
